sql-db/narayana-transactions/src/test/java/io/quarkus/ts/transactions/MssqlTransactionGeneralUsageIT.java

```diff
@@ -65,4 +72,22 @@ protected Operation[] getExpectedJdbcOperations() {
                 new Operation(actualOperationName -> actualOperationName.startsWith("UPDATE msdb.")) };
     }
 
+    @Test
+    @Tag("QUARKUS-4185")
+    //on average, there is one leaking connection every 2 minutes
+    void connectionLeak() throws InterruptedException {
+        int before = getConnections();
+        Duration later = Duration.of(4L, ChronoUnit.MINUTES).plus(Duration.ofSeconds(15));
+        Thread.sleep(later.toMillis());
```

Comment:
I wonder if adding logging like "going to sleep" would help when reading logs. It's purely optional.

Reply:
+1 to add such a message

Reply:
Added the log